python-frontend/src/main/java/org/sonar/python/tree/PythonTreeMaker.java

```diff
@@ -822,9 +823,31 @@ public Guard guard(AstNode guardNode) {
     return new GuardImpl(ifKeyword, condition);
   }
 
-  private static List<Pattern> patterns(AstNode pattern) {
-    // TODO: consider OR Patterns and other kind of patterns
-    AstNode literalPattern = pattern.getFirstChild(PythonGrammar.LITERAL_PATTERN);
+  public static Pattern pattern(AstNode pattern) {
+    if (pattern.is(PythonGrammar.CLOSED_PATTERN)) {
+      return closedPattern(pattern);
+    }
+    return asPattern(pattern);
+  }
+
+  private static AsPattern asPattern(AstNode asPattern) {
+    Pattern pattern = literalPattern(asPattern.getFirstChild(PythonGrammar.CLOSED_PATTERN).getFirstChild());
```

Comment:
Does that work for `case x as y:`? (I'm not sure that's valid)

Reply:
Yes , I think it's valid and it works after commit 782042ffdfa6371194397ac50b955dfb6f7d625d. I'll add a test

Reply:
Did you add a test? As we call `literalPattern`, I don't get how it works...

Reply:
Sorry, I misinterpreted your comment. I added a test in [`CasePatternsTest`](https://github.com/SonarSource/sonar-python/blob/22336e3e8b5fccd5491964c6dbf0597095a4dc92/python-frontend/src/test/java/org/sonar/python/parser/compound/statements/CasePatternsTest.java#L75) for the low level. grammar. However the problem you mentioned happens during the strongly typed AST translation.

I will treat this in a separate PR